DOL.WHD.Section14c.Web/src/modules/components/attachmentField/attachmentFieldController.js

```diff
@@ -12,43 +12,74 @@ module.exports = function(ngModule) {
     var vm = this;
     vm.stateService = stateService;
     vm.apiService = apiService;
+    $scope.restrictUpload = false;
+    if(!$scope.attachmentId) {
+       vm.upload = {
+        status: "NoFile",
+        message: 'No file is selected.'
+      }
+    } else {
+      vm.upload = {
+        status: "Success",
+        message: 'File was uploaded successfully.'
+      }
+    }
+
+    this.allowedFileTypes = ['pdf', 'jpg', 'JPG', 'jpeg', 'JPEG', 'png', 'PNG', 'csv', 'CSV', 'PDF'];
 
     this.onAttachmentSelected = function(fileinput) {
-      if (fileinput.files.length > 0) {
-        apiService
-          .uploadAttachment(
-            stateService.access_token,
-            stateService.ein,
-            fileinput.files[0]
-          )
-          .then(
-            function(result) {
-              $scope.attachmentId = result.data[0].id;
-              $scope.attachmentName = result.data[0].originalFileName;
-              fileinput.value = '';
-            },
-            function() {
-              //TODO: Display error
-              fileinput.value = '';
-            }
-          );
+      vm.upload.status = "Uploading";
+      vm.upload.message = 'File is uploading.'
+      if (fileinput.files.length > 0 ) {
+        vm.validateAttachment(fileinput.files[0], vm.allowedFileTypes);
+        vm.uploadAttachment(fileinput);
+      }
+    };
+
+    this.validateAttachment = function (fileinput, allowedFileTypes) {
+      var ext = fileinput.name.split(".")[1];
+      if(allowedFileTypes.indexOf(ext) < 0) {
+        vm.upload.status = 'Invalid';
+        vm.upload.message = 'Invalid File Type.';
+        return false;
+      }
+      if (fileinput.size / 1024000 > 5) {
```

Comment:
Not sure if this is supposed to be a megabyte, but that's 1,048,576.  That's a tiny nitpick and please don't change it because it's not worth the effort.  😛 

That said, should this maximum size be in `env.js` so it's more configurable?